abdiff/core/utils.py

```diff
@@ -4,35 +4,47 @@
 import os
 from pathlib import Path
 
-from slugify import slugify
-
 from abdiff.config import Config
 
 CONFIG = Config()
 
 
-def get_job_slug_and_working_directory(job_name: str) -> tuple[str, Path]:
-    """Create working directory for new job by slugifying job name."""
-    job_slug = slugify(job_name)
-    return job_slug, Path(CONFIG.root_working_directory) / job_slug
+def read_job_json(job_directory: str) -> dict:
+    """Read job JSON file."""
+    with open(Path(job_directory) / "job.json") as f:
+        return json.load(f)
+
+
+def read_run_json(run_directory: str) -> dict:
+    """Read run JSON file."""
+    with open(Path(run_directory) / "run.json") as f:
+        return json.load(f)
+
+
+def update_or_create_json(
+    directory: str | Path,
+    filename: str,
+    new_data: dict,
+) -> dict:
+    filepath = Path(directory) / filename
+    data = {}
+
+    if os.path.exists(filepath):
+        with open(filepath) as f:
+            data = json.load(f)
+    data.update(new_data)
 
+    with open(filepath, "w") as f:
+        json.dump(data, f, indent=2)
```

Comment:
Just to confirm, since we're using file `mode="w"`, this method will always read in the entire content of the JSON into a dictionary, update the dictionary with `new_data`, and then overwrite the existing JSON file, right?

Reply:
Yep!